src/Ast.ml

```diff
@@ -197,6 +197,28 @@ let doc_atrs atrs =
   let docs = match docs with [] -> None | l -> Some (List.rev l) in
   (docs, List.rev rev_atrs)
 
+let longident_is_simple (c : Conf.t) x =
+  let rec length (x : Longident.t) =
+    match x with
+    | Lident x -> String.length x
+    | Ldot (x, y) -> length x + 1 + String.length y
+    | Lapply (x, y) -> length x + length y
+  in
+  length x * 3 < c.margin
```

Comment:
Why `* 3` ?

Reply:
Note that such a criterion for "simple" is also used elsewhere, and IIRC a third of the margin is used everywhere. We could make the ratio configurable, it is arbitrary to a degree. The original thinking behind it is that it is small enough that only in very extreme cases will the indentation be 2/3 or the margin. Why a third instead of a half or whatever is a good question, and if some empirical evaluation was done that would be good. FTR I tried 1/2 and it seemed to allow some expressions which were long enough that grouping them together in wrapped application arguments started to get potentially hard to read, so 1/3 seemed better.

Reply:
The context here is somewhat different, as we do not expect multiple simple e.g. module aliases on a line, so perhaps something more like 2/3 would be better, to remain basically insensitive to the starting indentation.

Reply:
I understand. I'm still not convinced that this is a good idea. It seems too arbitrary and someone may report this as a bug later.

For the code, this must be documented. Also, I think it would be more understandable if the maths were done on the rhs (`length x < c.margin / 3`).

Reply:
I agree it's not ideal but at least it's consistent with the rest of the code. I added new functions and comments for consistency.